starter-code/src/components/ListOfCountries.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+class ListOfCountries extends Component {
+  constructor(props) {
+    super(props);
+    this.makeListOfCountries = this.makeListOfCountries.bind(this);
+  }
+
+  makeListOfCountries = () => {
```

Comment:
Not needed to create a method for generating the list of countries. You can also assign the `.map(... ` directly to a variable and cal that variable in the `return()`. See example below: 

```
import React, { Component } from "react";
import { Link } from "react-router-dom";

class ListOfCountries extends Component {
  render() {
    let listOfCountriesArr = this.props.countriesInfo.map((country, index) => {
      return (
        <Link
          key={index}
          className="list-group-item list-group-item-action"
          to={{
            pathname: `/about/${country.cca3}`,
            state: {
              borders: country.borders,
              countryName: country.name.common,
              capital: country.capital,
              area: country.area,
              countryNameArr: this.props.makeCountryNameArr(country.borders),
              listOfCountries: this.props.countriesInfo
            }
          }}
        >
          <span>{country.flag}</span>
          {country.name.common}
        </Link>
      );
    });

    return (
      <div className="col-5" style={{ maxHeight: "90vh", overflow: "scroll" }}>
        <div className="list-group">{listOfCountriesArr}</div>
      </div>
    );
  }
}

export default ListOfCountries;

```